scheduler/scheduler-server/src/test/java/performancetests/recovery/NodeRecoveryTest.java

```diff
@@ -25,37 +25,117 @@
+import java.util.*;
 
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.objectweb.proactive.core.config.ProActiveConfiguration;
+import org.ow2.proactive.resourcemanager.common.RMConstants;
+import org.ow2.proactive.resourcemanager.core.RMCore;
+import org.ow2.proactive.resourcemanager.core.properties.PAResourceManagerProperties;
+import org.ow2.proactive.resourcemanager.frontend.ResourceManager;
 
-public class NodeRecoveryTest extends AbstractJavaSamplerClient {
+import functionaltests.nodesrecovery.RecoverInfrastructureTestHelper;
+import functionaltests.utils.*;
+import performancetests.helper.LogProcessor;
 
-    @Override
-    public SampleResult runTest(JavaSamplerContext javaSamplerContext) {
-        SampleResult sampleResult = null;
-        try {
-            final Integer nodesNumber = Integer.valueOf(JMeterContextService.getContext()
-                                                                            .getVariables()
-                                                                            .get("nodesNumber"));
 
-            NodeRecoveryHelper nodeRecoveryHelper = new NodeRecoveryHelper();
+/**
+ * @author ActiveEon Team
+ * @since 01/12/17
+ */
+@RunWith(Parameterized.class)
+public class NodeRecoveryTest extends SchedulerFunctionalTestWithCustomConfigAndRestart {
+
+    static final String RM_CONFIGURATION_START = NodeRecoveryTest.class.getResource("/performancetests/config/rm-start.ini")
+                                                                       .getPath();
+
+    static final String RM_CONFIGURATION_RESTART = NodeRecoveryTest.class.getResource("/performancetests/config/rm-restart.ini")
+                                                                         .getPath();
+
+    @Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(new Object[][] { { 5, 1000 }, { 100, 2000 }, { 200, 1000 }, { 300, 3000 } });
+    }
+
+    // number of nodes
+    int nodesNumber;
+
+    // time limit in milliseconds for test to pass
+    int timeLimit;
+
+    public NodeRecoveryTest(int nodesNumber, int timeLimit) {
+        this.nodesNumber = nodesNumber;
+        this.timeLimit = timeLimit;
+    }
+
+    /**
+     * This method tests performance of node recovery using local infrastructure.
+     *
+     * @throws Exception
+     */
+    @Before
+    public void startKillStartRM() throws Exception {
+        ProActiveConfiguration.load();
+        RMTHelper rmHelper = new RMTHelper();
+        rmHelper.startRM(RM_CONFIGURATION_START);
+
+        assertTrue(PAResourceManagerProperties.RM_PRESERVE_NODES_ON_SHUTDOWN.getValueAsBoolean());
+        assertTrue(rmHelper.isRMStarted());
+
+        ResourceManager resourceManager = rmHelper.getResourceManager();
+
+        assertEquals(resourceManager.getState().getAllNodes().size(), 0);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Swap these 2 arguments so they are in the correct order: expected value, actual value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3415)